ffi/capi/src/fixed_decimal.rs

```diff
@@ -15,13 +16,34 @@ pub type ICU4XFixedDecimal = FixedDecimal;
 #[no_mangle]
 /// FFI version of [`FixedDecimal`]'s constructors. This constructs a [`FixedDecimal`] of the provided
 /// `number`.
-//
-// We can add additional constructors from strings, floats, etc as the need arises
 pub extern "C" fn icu4x_fixed_decimal_create(number: i64) -> *mut ICU4XFixedDecimal {
     let fd = FixedDecimal::from(number);
     Box::into_raw(Box::new(fd))
 }
 
+#[no_mangle]
+/// FFI version of [`FixedDecimal::FromStr()`], see its docs for more details
+///
+/// # Safety
+/// `value` and `len` should point to a valid ASCII string of length `len`.
+///
+/// It does not need to be be null terminated, and `len` should not include a null
+/// terminator (this will just cause the function to panic, and is not a safety requirement).
+///
+/// Returns nullptr if passed an invalid string.
```

Comment:
This seems Resulty, not Optiony, so we should probably use a custom Result type with a boolean success value (like we do elsewhere) that can later be turned into an enum, unless we are sure we want this to always stay Optionlike.


Ultimately the C API is not supposed to be idiomatic, it's supposed to be wrappable with idiomatic APIs in all languages.